### PR TITLE
Konsti papyrus recording

### PR DESCRIPTION
**Pull request by KonstiNik**
In this PR the JaxRecorder is replaced by a recorder implementation in the package papyrus. 
For that the following changes will be applied:

- [x] Create a class for Recording with the papyrus package, see JaxRecorder in _znnl.training_recording.papyrus_jax_recording.py_.
- [x] Move the NTK computation from a model attribute to an independent class in the analysis module: JAXNTKComputation
- [x] Create NTK sampling strategies: JAXNTKSubsampling, JAXNTKClassWise, JAXNTKCombinations

### JAXNTKComputation
Re-implementation of the previous NTK computation located in each model.  
The now has to set an apply function to construct the class. This allows for setting any function on which the NTK should be recorded. 

### JAXNTKSubsampling
An NTK computation that approximates the full matrix by computing diagonal blocks of the NTK using user-defined block sizes. The block size can be set via the arg `ntk_size` when constructing the method. The data of each block matrix is assigned randomly. 

### NTKClassWise
An NTK computation that computes kernel for all samples of the same class. Given a data set of 10 classes, one obtains 10 NTKs. 

### NTKCombinations
An NTK computation that evaluates returns the Kernel for all possible class combinations. Given 2 classes, one obtains the NTK for the samples of class (0), (1) and (0+1). Which classes to be selected can be controlled at construction by setting the arg `class_labels`.

### CODE REVIEW

File: .github/workflows/nbtest.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ jobs:
             pip3 install nbmake
         - name: Install package
           run: |
+            git config --global url."https://${{ secrets.PAPYRUS_ACCESS }}@github".insteadOf https://github
```
Comment (SamTov) [inline .github/workflows/nbtest.yml:21]: We can just make it public.

**Review by SamTov — APPROVED** [review]
The papyrus addition looks good. The additional recorders could have been put into a separate PR, but I read over them.

**Review by SamTov — APPROVED** [review]
Would be better to split the PRs but it looks good.

File: .github/workflows/nbtest.yml (hunk 1 above)
Comment (KonstiNik) [reply .github/workflows/nbtest.yml:21]: Just made it public.

Comment (KonstiNik) [reply .github/workflows/nbtest.yml]: Included in [a27cdac](https://github.com/zincware/ZnNL/pull/121/commits/a27cdace065c2a59e12574bf57fdc39370782e21)

**Pull request merged**